File: GitToolBox/src/test/java/zielu/gittoolbox/metrics/MockMetrics.java
package zielu.gittoolbox.metrics;

import com.codahale.metrics.Counter;
import com.codahale.metrics.Gauge;
import com.codahale.metrics.Timer;
import org.jetbrains.annotations.NotNull;

public class MockMetrics implements AppMetrics, ProjectMetrics {
  private Metrics metrics = new MetricManager();

  @Override
  public Timer timer(@NotNull String simpleName) {
    return metrics.timer(simpleName);
  }

  @Override
  public Counter counter(@NotNull String simpleName) {
    return metrics.counter(simpleName);
  }

  @Override
  public <T> Gauge<T> gauge(@NotNull String simpleName, Gauge<T> gauge) {
    return metrics.gauge(simpleName, gauge);
  }
}

File: GitToolBox/src/test/kotlin/zielu/gittoolbox/metrics/MockMetrics.kt
package zielu.gittoolbox.metrics

import zielu.intellij.metrics.GtCounter
import zielu.intellij.metrics.GtGauge
import zielu.intellij.metrics.GtTimer
import zielu.intellij.metrics.codehale.CodehaleMetricsManager

internal class MockMetrics : AppMetrics, ProjectMetrics {
  private val metrics = CodehaleMetricsManager()

  override fun startReporting() {
    // do nothing
  }

  override fun timer(simpleName: String): GtTimer = metrics.timer(simpleName)

  override fun counter(simpleName: String): GtCounter = metrics.counter(simpleName)

  override fun <T : Any?> gauge(simpleName: String, value: () -> T): GtGauge {
    return metrics.gauge(simpleName, value)
  }
}
